helix-core/src/test/java/org/apache/helix/controller/stages/BaseStageTest.java

```diff
@@ -209,7 +210,11 @@ protected void runStage(ClusterEvent event, Stage stage) {
     stage.init(context);
     stage.preProcess();
     try {
-      stage.process(event);
+      if (stage instanceof AbstractAsyncBaseStage) {
+        ((AbstractAsyncBaseStage) stage).execute(event);
```

Comment:
Why we need this?
 Do we have test for this?

Reply:
This is necessary to execute the business logic of any async stage because that logic is in execute method instead of process method since AbstractAsyncStage overrides the process method to run execute on async worker. TestAsyncBaseStage tests AbstractAsyncStage's process method.

Reply:
Tests for business logic for TopStateHandoffReportStage is in TopStateHandoffReportMetrics.